client/sources/ok_test/models.py

```diff
@@ -179,15 +180,22 @@ def dump(self):
         with open(test_tmp, 'w', encoding='utf-8') as f:
             f.write('test = {}\n'.format(json))
 
-        # Try to use os.replace, but if on Windows manually remove then rename
-        # (ref issue #339)
-        if os.name == 'nt':
-        # TODO(colin) Add additional error handling in case process gets killed mid remove/rename
-            os.remove(self.file)
-            os.rename(test_tmp, self.file)
-        else:
-            # Use an atomic rename operation to prevent test corruption
-            os.replace(test_tmp, self.file)
+        success = False
+        try:
+            storage.replace_transactional(test_tmp, self.file)
+            success = True
+        except (NotImplementedError, OSError):
+            pass
+        if not success:
```

Comment:
Would it make sense for this block to be inside the body of the `except` clause and the `success = False; success = True` bit go away.

Reply:
`ok-client` should be python3 only, but I'm curious about that issue. There are issues raising different exceptions in an except block in python2?

Reply:
Yup! What would Python print? :-)
```
try:
	try: raise ValueError("ValueError")
	except:
		try: raise NotImplementedError("NotImplementedError")
		except: pass
		raise
except Exception as ex: print(ex)
```

Reply:
Okay I just removed `success`, see the updated code.

Reply:
Oh I just realized I don't raise `NotImplementedError` properly... let me fix that.

Reply:
Ok fixed.